src/main/java/org/nrnb/gsoc/enrichment/tasks/FilterEnrichmentTableTask.java

```diff
@@ -16,92 +17,114 @@
-
-	public FilterEnrichmentTableTask(final CyServiceRegistrar registrar, EnrichmentCytoPanel panel) {
-		this.registrar = registrar;
-		applicationManager = registrar.getService(CyApplicationManager.class);
-		this.network = applicationManager.getCurrentNetwork();
-		this.enrichmentPanel = panel;
-	}
-
-
-
-	@Override
-	public void run(TaskMonitor monitor) throws Exception {
-		monitor.setTitle("Filter Enrichment table");
-		logger.info("Filtering results based on selected value: " + categories.getSelectedValues());
-
-		List<TermSource> categoryList = categories.getSelectedValues();
-		if (enrichmentPanel == null) {
-			CySwingApplication swingApplication = registrar.getService(CySwingApplication.class);
-			CytoPanel cytoPanel = swingApplication.getCytoPanel(CytoPanelName.SOUTH);
-			if (cytoPanel.indexOfComponent("org.nrnb.gsoc.enrichment") != -1)
-				enrichmentPanel = (EnrichmentCytoPanel) cytoPanel.getComponentAt(
-						cytoPanel.indexOfComponent("org.nrnb.gsoc.enrichment"));
-			else return;
-		}
-		EnrichmentTableModel tableModel = enrichmentPanel.getTableModel();
-		if (tableModel == null){
-			monitor.showMessage(TaskMonitor.Level.ERROR,  "Unable to find enrichment table!");
-		}
-		tableModel.filter(categoryList);
-		logger.info("Filtering results completed");
-	}
-
-	@Override
-	@SuppressWarnings("unchecked")
-	public <R> R getResults(Class<? extends R> clzz) {
-		if (clzz.equals(String.class)) {
-			if (filteredEnrichmentTable != null)
-				return (R)("\"EnrichmentTable\": "+filteredEnrichmentTable.getSUID());
-			return (R)"";
-		} else if (clzz.equals(JSONResult.class)) {
-			JSONResult res = () -> {
-				if (filteredEnrichmentTable != null)
-					return "{\"EnrichmentTable\": "+filteredEnrichmentTable.getSUID()+"}";
-				return "{}";
-			};
-			return (R)res;
-		}
-		return null;
-	}
-
-	@Override
-	public List<Class<?>> getResultClasses() {
-		return Arrays.asList(JSONResult.class, String.class);
-	}
-
-	@ProvidesTitle
-	public String getTitle() {
-		return "Filter Enrichment table";
-	}
+    private final Logger logger = Logger.getLogger(CyUserLog.NAME);
+
+    //public boolean removeOverlapping = false;
+
+    //public BoundedDouble overlapCutoff = new BoundedDouble(0.0, 0.5, 1.0, false, false);
+
+    public FilterEnrichmentTableTask(final CyServiceRegistrar registrar, EnrichmentCytoPanel panel) {
+        this.registrar = registrar;
+        applicationManager = registrar.getService(CyApplicationManager.class);
+        this.network = applicationManager.getCurrentNetwork();
+        this.enrichmentPanel = panel;
+        final CyColumn evidenceColumn = ModelUtils.getEnrichmentTable(registrar, network, TermSource.ALL.getTable())
+                .getColumn(EnrichmentTerm.colTermID);
+        System.out.println(evidenceColumn.getValues(String.class).size());
+
+        // TODO: Create EVIDENCE CODE final class as per GProfiler
+        final Set<String> evidenceCodeList = new TreeSet<>(evidenceColumn.getValues(String.class));
```

Comment:
Why TreeSet? Do we need the evidenceCodeList to be sorted?

Reply:
This section of code belongs to an older commit. We are now using values from the enum which are already sorted.